modules/katello_sync_plan.py

```diff
@@ -84,154 +96,89 @@
-            if hasattr(new, field) and hasattr(old, field):
-                new_attr = getattr(new, field)
-                old_attr = getattr(old, field)
-                if old_attr is None or (hasattr(new_attr, 'id') and hasattr(old_attr, 'id') and new_attr.id != old_attr.id) or new_attr != old_attr:
-                    setattr(old, field, new_attr)
-                    needs_update = True
-            elif hasattr(old, field) and getattr(old, field) is not None and not hasattr(new, field):
-                setattr(old, field, None)
-                needs_update = True
-        return needs_update, old
-
-    def sync_plan(self, name, organization, interval=None, enabled=None, sync_date=None, products=[]):
-        updated = False
-        organization = self.find_organization(organization)
-
-        enabled = enabled.lower() == "true"
-
-        sync_plan = self._entities.SyncPlan(self._server, name=name, organization=organization, interval=interval, enabled=enabled, sync_date=sync_date)
-        response = sync_plan.search({'name', 'organization'})
-
-        if len(response) == 1:
-            response[0].sync_date = parse_time(response[0].sync_date)
-            updated, sync_plan = self.update_fields(sync_plan, response[0], ['interval', 'enabled', 'sync_date'])
-            if updated and not self._module.check_mode:
-                sync_plan.update()
-        elif len(response) == 0:
-            if not self._module.check_mode:
-                sync_plan = sync_plan.create()
-            updated = True
-
-        desired_product_ids = list(map(lambda p: p.id, self.find_products(products, organization)))
-        current_product_ids = list(map(lambda p: p.id, sync_plan.product))
+RETURN = ''' # '''
+
+from ansible.module_utils.ansible_nailgun_cement import (
+    SyncPlan,
+    find_organization,
+    find_products,
+    find_sync_plan,
+    naildown_entity,
+    sanitize_entity_dict,
+)
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
 
-        if set(desired_product_ids) != set(current_product_ids):
-            if not self._module.check_mode:
-                product_ids_to_add = set(desired_product_ids) - set(current_product_ids)
-                if len(product_ids_to_add) > 0:
-                    sync_plan.add_products(data={'product_ids': list(product_ids_to_add)})
-                product_ids_to_remove = set(current_product_ids) - set(desired_product_ids)
-                if len(product_ids_to_remove) > 0:
-                    sync_plan.remove_products(data={'product_ids': list(product_ids_to_remove)})
-            updated = True
 
-        return updated
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'description': 'description',
+    'organization': 'organization',
+    'interval': 'interval',
+    'enabled': 'enabled',
+    'sync_date': 'sync_date',
+    'cron_expression': 'cron_expression',
+    'products': 'product',
```

Comment:
Huh? Isn't this supposed to be an array? Are we changing how to use this module?